### PR TITLE
Upgrade react-navigation

### LINKED ISSUE #316: Accessing menus in sidebar overlays previous detail screen while new screen animates in
[previous-screen-glitch.zip](https://github.com/HoardInvest/hoard/files/2764216/previous-screen-glitch.zip)
See attached videos

### PR DESCRIPTION
**Pull request by zachgibb**
Fixes #316

### CODE REVIEW

File: src/screens/Settings/Security/Security.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ export default class Security extends Component {
   };
 
   render() {
+    console.log('rendered');
```
Comment (ericgibb) [inline src/screens/Settings/Security/Security.js]: ✂️

File: src/components/Base/Navigation.js [hunk 2]
```diff
@@ -164,7 +164,7 @@ export const getNavigationOptions = navProps => {
 
 export const transitionConfig = () => {
   return {
-    containerStyle: { backgroundColor: 'transparent' },
+    containerStyle: {},
```
Comment (ericgibb) [inline src/components/Base/Navigation.js]: we could probably remove this whole obj

File: package.json [hunk 3]
```diff
@@ -86,7 +87,7 @@
     "react-native-swipeable": "^0.6.0",
     "react-native-touch-id": "4.1.1",
     "react-native-vector-icons": "^4.6.0",
-    "react-navigation": "2.9.3",
+    "react-navigation": "^3.1.5",
```
Comment (ericgibb) [inline package.json]: lock to specific version here?

File: src/components/Base/Navigation.js (hunk 2 above)
Comment (zachgibb) [reply src/components/Base/Navigation.js]: nah - unfortunately we need to override the default object entirely, which means we need to pass in our own object even if it's empty.

when we transition to a solid color, this issue will be resolved anyway.

File: src/screens/Settings/Security/Security.js (hunk 1 above)
Comment (zachgibb) [reply src/screens/Settings/Security/Security.js]: snipped

File: package.json (hunk 3 above)
Comment (zachgibb) [reply package.json]: locked

**Pull request merged**